### PR TITLE
Optimize computed properties output (byte-wise)

### PR DESCRIPTION
**Pull request by Andarist**
| Q                        | A 
| ------------------------ | ---
| Fixed Issues?            | n/a
| Patch: Bug Fix?          | no
| Major: Breaking Change?  | no
| Minor: New Feature?      | no
| Tests Added + Pass?      | Yes
| Documentation PR         | no
| Any Dependency Changes?  | no

It is a small "optimization" of the computed properties output. The change basically wraps (if possible) previous element - `initProps` object or previous `defineProperty` call when "pushing" new `defineProperty` calls.

It leverages the fact that `defineProperty` returns the passed object, therefore in most cases we do not need to create temporary references and stuff.

[babel-plugin-transform-computed-properties/test/fixtures/spec/multiple](https://github.com/babel/babel/compare/master...Andarist:polish/computed-properties?expand=1#diff-226ea89fff707b39966cfbb41e57eaa8) illustrates this nicely.

**Before**
```js
var _obj;
var obj = (_obj = {}, babelHelpers.defineProperty(_obj, "x" + foo, "heh"), babelHelpers.defineProperty(_obj, "y" + bar, "noo"), _obj);
```

**After**
```js
var obj = babelHelpers.defineProperty(babelHelpers.defineProperty({}, "x" + foo, "heh"), "y" + bar, "noo");
```

Not only it saves some bytes in the output, it is probably more friendly for the minifiers too.

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/6652"></a>

### CODE REVIEW

File: packages/babel-plugin-transform-computed-properties/test/fixtures/spec/mixed/actual.js [hunk 1]
```diff
@@ -1,6 +1,11 @@
+var heh = "heh";
+var noo = "noo";
+var foo = "foo";
+
 var obj = {
```
Comment (xtuc) [inline packages/babel-plugin-transform-computed-properties/test/fixtures/spec/mixed/actual.js]: I don't understand how it worked before, either `foo` and `bar` were undefined in the score or you now re-defined them?

Comment (Andarist) [reply packages/babel-plugin-transform-computed-properties/test/fixtures/spec/mixed/actual.js]: This was purely a fixture/snapshot test, I've added also `exec.js` and that's why I had to add those to the scope in actual/expected fixtures (like you have noticed - they were previously undefined).

Comment (xtuc) [reply packages/babel-plugin-transform-computed-properties/test/fixtures/spec/mixed/actual.js]: Ok that make sense

**Comment by xtuc** [comment]
@Andarist do we stil want this change?

**Comment by Andarist** [comment]
I would be in favour of merging this (after rebase ofc), but unfortunately noone (except you) has reviewed this.

**Comment by xtuc** [comment]
My review wasn't very useful, to push this forward we could discuss about it during our next meeting. Could you please add it to our agenda here https://github.com/babel/notes/issues/73. thanks

**Comment by nicolo-ribaudo** [comment]
@JLHwung @liuxingbaoyu This is something that we can still do. I rebased this PR, and added a cap as suggested in https://github.com/babel/babel/pull/6652#discussion_r234700582.

**Comment by Andarist** [comment]
Landing this PR was not my 2023 bingo card… 😝

**Review by JLHwung — APPROVED** [review]

**Review by nicolo-ribaudo — APPROVED** [review]

**Pull request merged**